docs/developers-docs/development-environment/frontend_environment.md

```diff
@@ -0,0 +1,160 @@
+infrastructure use the following `config/database.yml`
+
+```yml
+    development:
+      adapter: mysql2
+      encoding: utf8
+      database: bioportal_ui
+      username: root
+      password: bp_user
+      host: db
+```
+
+### Update the development.rb file
+
+{: .highlight }
+A sample file of `config/environments/development.rb` can be found
+here:[https://github.com/ontoportal/ontoportal_web_ui/blob/master/config/environments/development.rb](https://github.com/ontoportal/ontoportal_web_ui/blob/master/config/environments/development.rb)
+
+Update like below the line in `config/environments/development.rb` file
+
+```ruby
+ # memcache setup
+config.cache_store = ActiveSupport::Cache::MemCacheStore.new('cache', namespace: 'BioPortal')
+
+```
+
+### Create bioportal_config_development.rb file
+
+{: .highlight }
+A sample file of `config/bioportal_config_development.rb` can be found
+here:[https://github.com/ontoportal/ontoportal_web_ui/blob/master/config/bioportal_config_env.rb.sample](https://github.com/ontoportal/ontoportal_web_ui/blob/master/config/bioportal_config_env.rb.sample)
+
+```bash
+cp .env.sample .env
+
+cp bioportal_config_env.rb.sample bioportal_config_development.rb
+```
+
+Update/Add the following lines to `config/bioportal_config_development.rb`  
+
+```ruby
+
+$HOSTNAME = 'localhost' # PUT YOUR SERVER HOSTNAME (IP) HERE
+
+# The URL for the BioPortal Rails UI (this application)
+$UI_URL = "http://#{$HOSTNAME}:3000"
+
+# REST core service address
+$REST_URL = "http://#{$HOSTNAME}:8080"
+$SPARQL_URL = "http://#{$HOSTNAME}:8081/test/"
+
+$BIOMIXER_URL = "http://#{$HOSTNAME}:8081/BioMixer"
+
+$ANNOTATOR_URL = $PROXY_URL = "http://#{$HOSTNAME}:8081/annotator"
+
+# If your BioPortal installation includes Fairness score set this to true
+$FAIRNESS_DISABLED = false
+$FAIRNESS_URL = "http://#{$HOSTNAME}:8081/fairness"
+
+# NCBO annotator URL and apikey
+$NCBO_ANNOTATOR_URL = "http://#{$HOSTNAME}:8081/ncbo_annotatorplus"
+$NCBO_ANNOTATORPLUS_ENABLED = true
+$NCBO_API_KEY = "4a5011ea-75fa-4be6-8e89-f45c8c84844e"
+
+# Unique string representing the UI's id for use with the BioPortal Core
+$API_KEY = "72c72cba-ad45-4785-b94e-483fa55cdddb"
+
+```
+
+## Build the docker application
+
+{: .highlight }
+Source code of the docker infrastructure can be found
+here: [Feature: Add docker environment for the UI ncbo/bioportal_web_ui#229](https://github.com/ncbo/bioportal_web_ui/pull/229)
+
+Install docker and docker-compose,
+see [https://docs.docker.com/compose/install/](https://docs.docker.com/compose/install/)
+
+Build everything: The first time you run this it's going to take 5-10 minutes depending on your internet connection
+speed and computer's
+hardware specs. That's because it's going to download a few Docker images and build the Ruby + Yarn dependencies (1,5 GB
+for all).
+
+```bash
+docker compose build
+```
+
+## Run the local development server
+
+Run the docker environment
+
+```bash
+docker compose up
+```
+
```

Comment:
I would also mention the one-time command:
docker compose exec web sh -c "bin/setup"